### DISCUSSION TITLE
Amarisoft enb test

### DISCUSSION THREAD

**Discussion by arrowtem** [srsran/srsRAN_Project#260] [General Help] [answered]
Developers of SRSRAN told that they were testing srsENB with 64 active UE’s connected 
I got some questions because of it 
1) How is it possible to run 1 on sdr multiple UE’s , If their positions of PUCCH resources could overlap in freq domain ( the same with RACH) 
2) Were you testing 64 UE’s connected and performing traffic test ? IPerf? Was enb able to handle it ?
Thanks much for answer !

**Comment by pgawlowicz** [comment, marked as answer]
AmarisoftUE emulates multiple UEs. Specifically, it generates UL signals from each of them, combines all signals, and transmits the final signal through a single SDR radio. Note that PUCCH and RACH issues are the same in the case of independent (_normal_) UEs. It is up to a gnb to overcome those issues (e.g., contention resolution mechanism in RACH). 

Yes, we test with 32 and 64 UEs, and the gnb handles it without issues.
